apps/wallet-dashboard/hooks/useFilterMigrationObjectDetails.ts

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+import { MIGRATION_OBJECT_WITHOUT_UC_KEY } from '@/lib/constants';
+import { CommonMigrationObjectType, StardustOutputDetailsFilter } from '@/lib/enums';
+import { ResolvedObjectTypes } from '@/lib/types';
+import { useMemo } from 'react';
+
+export function useFilterMigrationObjects(
+    objects: ResolvedObjectTypes[],
+    filter: StardustOutputDetailsFilter,
+) {
+    return useMemo(() => {
```

Comment:
```suggestion
```

Reply:
I disagree here because the list could potentially be very big and redoing the filter in each render could be worse than storing it in a memo

Reply:
I disagree with your disagreement, I think the common approach is to only optimize (useMemo) if you see some performance issues

That said, I leave it up to you to decide